### DISCUSSION TITLE
Typescript+pinia: Problem about type of this

### DISCUSSION THREAD

**Discussion by YornQiu** [vuejs/pinia#1172] [Help and Questions] [answered]
I created a store, it's like:
```
import { defineStore } from 'pinia';

export interface Person {
  name: string;
  age: number;
}

export default defineStore({
  id: 'useFactorFilters',
  state: (): Person => ({
    name: 'Jack',
    age: 18,
  }),
  actions: {
    setInfo<T extends keyof Person>(key: T, value: Person[T]) {
      this[key] = value;
    },
  },
});
```
Function setInfo is ok when it's  used alone in other situations. But in pinia , i got a error:
![image](https://user-images.githubusercontent.com/13913084/160171141-aef51531-3781-42cb-9006-4556486155e2.png)

I don't know how to handle it.
Appreciate if any suggestion!

And it only comes when the keys of Person are different types.

**Comment by posva** [comment, marked as answer]
You can do

```ts
this.$state[key] = value
```

But I would recommend you not to create an action if it just sets a value:

```ts
store.name = newName
store.age = newAge
```

**Reply by YornQiu** [reply]
Got it. 
Thx!
